staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher_whitebox_test.go

```diff
@@ -2396,7 +2396,7 @@ func TestWatchStreamSeparation(t *testing.T) {
 				defer cacher.watchCache.RUnlock()
 				return cacher.watchCache.resourceVersion
 			}
-			waitContext, cancel := context.WithTimeout(context.Background(), 2*time.Second)
+			waitContext, cancel := context.WithTimeout(context.Background(), time.Second)
```

Comment:
this only matters when something goes wrong and the fails, right ?

Reply:
Right.

Reply:
agreed, this is the upper-bound we will wait for before failing the test by returning a blank RV. I don't think shortening this will actually make the test pass faster, just increase the possibility of failing

Reply:
It will because there are subtests where we expect to not get bookmark.

Reply:
ah, I see. are we confident a second is long enough to not flake on the tests where we *do* expect a bookmark event?

Reply:
> It will because there are subtests where we expect to not get bookmark.

Actually, if we choose our wait time differently for these two types of tests, we can leave the wait as-is (2 seconds) when we *do* expect to receive a bookmark, and shorten the wait more aggressively (500ms? 250ms?) when we *don't* expect to receive a bookmark.

Reply:
Isn't such asymmetry make the test less accurate? I understand that we can lower the timeout when we don't expect the bookmark as we don't need to wait, however on the other hand it makes it easy to miss if a regression is introduce later. In my opinion making a comparison test like this, we should keep the setup symmetrical to be sure.

Reply:
I don't think we'd lower it so much it would be easy to miss a regression. We should pick a time large enough that consistently would receive an incorrect bookmark event and fail the test.